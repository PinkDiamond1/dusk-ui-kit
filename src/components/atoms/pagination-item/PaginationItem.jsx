/* @flow */
import * as React from 'react'
import BootstrapPaginationItem from 'react-bootstrap/PageItem'

import './PaginationItem.scss'

type Props = {
  className: string,
  active: boolean,
  activeLabel: string,
  children: React.Node,
}

const PaginationItem = (props: Props): React.Element<*> => {
  const { className, active, activeLabel, children } = props
  const updatedClassName = `__duk-pagination-item ${className}`
  return (
    <BootstrapPaginationItem active={active} activeLabel={activeLabel} className={updatedClassName}>
      {children}
    </BootstrapPaginationItem>
  )
}

PaginationItem.defaultProps = {
  className: '',
  active: false,
  activeLabel: 'current',
}

export default PaginationItem
