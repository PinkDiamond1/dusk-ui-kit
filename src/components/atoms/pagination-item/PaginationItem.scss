.__duk-pagination-item.page-item {
  .page-link{
    border-radius: 5px;
  }
}
