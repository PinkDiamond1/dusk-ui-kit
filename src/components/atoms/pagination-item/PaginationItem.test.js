import React from 'react'
import renderer from 'react-test-renderer'

import PaginationItem from './'

it('test to see if the PaginationItem renders correctly', () => {
  const tree = renderer.create(<PaginationItem>Smoke Test</PaginationItem>).toJSON()
  expect(tree).toMatchSnapshot()
})
