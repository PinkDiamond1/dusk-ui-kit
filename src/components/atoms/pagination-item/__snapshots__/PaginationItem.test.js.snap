// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to see if the PaginationItem renders correctly 1`] = `
<li
  className="__duk-pagination-item  page-item"
>
  <a
    className="page-link"
    href="#"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
  >
    Smoke Test
  </a>
</li>
`;
